addons/website_sale/models/product.py

```diff
@@ -339,6 +342,21 @@ def _get_current_company_fallback(self, **kwargs):
         website = self.website_id or kwargs.get('website')
         return website and website.company_id or res
 
+    def _init_column(self, column_name):
+        # to avoid generating a single default website_sequence when installing the module,
+        # we need to set the default row by row for this column
+        if column_name == "website_sequence":
+            _logger.debug("Table '%s': setting default value of new column %s to unique values for each row", self._table, column_name)
+            self.env.cr.execute("SELECT id FROM %s WHERE website_sequence IS NULL" % self._table)
+            prod_tmpl_ids = self.env.cr.dictfetchall()
+            prod_tmpl_num = len(prod_tmpl_ids)
+            max_seq = self._default_website_sequence()
+            query_list = [{'id': prod_tmpl_id['id'], 'website_sequence': max_seq + (prod_tmpl_num - prod_tmpl_id['id']) * 5} for prod_tmpl_id in prod_tmpl_ids]
+            query = 'UPDATE ' + self._table + ' SET website_sequence = %(website_sequence)s WHERE id = %(id)s;'
+            self.env.cr._obj.executemany(query, query_list)
+        else:
+            super(ProductTemplate, self)._init_column(column_name)
+
     def _default_website_sequence(self):
```

Comment:
Can't we just loop on `self` in this function to assign correct unique sequence?

Reply:
As discussed, I think self is an empty recordset, so my suggestion is not of application :)